### PR TITLE
feat: announcement-ui as webcomponent

### PR DESCRIPTION
**Pull request by markuczy**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -74,13 +75,15 @@
...
+          "defaultConfiguration": "development",
           "builder": "ngx-build-plus:dev-server",
           "options": {
             "browserTarget": "onecx-announcement-ui:build",
             "port": 4200,
             "publicHost": "http://localhost:4200",
             "extraWebpackConfig": "webpack.config.js",
-            "proxyConfig": "proxy.conf.js"
+            "proxyConfig": "proxy.conf.js",
+            "disableHostCheck": true
```
Comment (markuczy) [inline angular.json:86]: remove

File: angular.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
               "node_modules/@onecx/portal-integration-angular/assets/output.css",
               "src/styles.scss"
             ],
-            "scripts": [],
+            "scripts": ["node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js"],
+            "outputHashing": "all",
```
Comment (markuczy) [inline angular.json:39]: remove?

File: src/app/remotes/announcement-banner-webcomponent/announcement-banner-webcomponent.component.bootstrap.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  }
+  const version = VERSION
+  platform && platformCache.set(version, platform)
+
+  const myStandaloneComponentAsWebComponent = createCustomElement(OneCXAnnouncementWebcomponentBannerComponent, {
+    injector: app.injector
+  })
+
+  customElements.define('ocx-announcement-banner-component', myStandaloneComponentAsWebComponent)
+})()
```
Comment (markuczy) [inline src/app/remotes/announcement-banner-webcomponent/announcement-banner-webcomponent.component.bootstrap.ts:37]: move to function

File: src/app/router.utils.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    const mfeBaseHref: string = route.data!['mfeInfo'].baseHref
+    const baseHrefSegmentAmount = mfeBaseHref.split('/').filter((value) => value).length
+    const urlWithoutBaseHref = url.slice(baseHrefSegmentAmount)
+    const index = urlWithoutBaseHref && prefix === '' ? 0 : urlWithoutBaseHref.findIndex((u) => u.path === prefix)
+    if (index >= 0) {
+      return { consumed: url.slice(0, index + baseHrefSegmentAmount + 1) }
+    }
+    return null
+  }
+}
```
Comment (markuczy) [inline src/app/router.utils.ts:13]: move to libs

File: src/bootstrap.ts [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+// if (environment.production) {
+//   enableProdMode()
+// }
 
-platformBrowserDynamic()
-  .bootstrapModule(AppModule)
-  .catch((err) => console.error(err))
+// platformBrowserDynamic()
+//   .bootstrapModule(AppModule)
+//   .catch((err) => console.error(err))
```
Comment (markuczy) [inline src/bootstrap.ts:13]: remove comments

File: webpack.config.js [hunk 6]
```diff
@@ -46,5 +44,17 @@ const plugins = config.plugins.filter((plugin) => !(plugin instanceof ModifyEntr
...
+    publicPath: 'auto'
+  },
+  experiments: {
+    ...config.experiments,
+    topLevelAwait: true
+  },
+  optimization: {
+    runtimeChunk: false,
+    splitChunks: false
+  }
```
Comment (markuczy) [inline webpack.config.js:59]: needed?

File: src/app/onecx-announcement-remote-webcomponent.module.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            route.data = {
+              ...route.data,
+              mfeInfo: mfeInfo
+            }
+          })
+          router.resetConfig(routes)
+        })
+      )
+    )
+}
```
Comment (markuczy) [inline src/app/onecx-announcement-remote-webcomponent.module.ts:39]: move to function

File: angular.json (hunk 1 above)
Comment (markuczy) [reply angular.json:86]: needed in order to debug ui code in browser

File: webpack.config.js (hunk 6 above)
Comment (markuczy) [reply webpack.config.js:57]: topLevelAwait -> needed to transform module into async module
runtimeChunk -> ?
splitChunks -> ?

**Review by SchettlerKoehler — APPROVED** [review]

**Pull request merged**